Updating Daemon Code

rmkraus opened the pull request:
Updating the daemon scripts to work more cleanly and work with the new distribution policies.

rmkraus committed:
Fixed importing of dependency for map tracker.

rmkraus committed:
Fixed device tracker to install scanner requirements.

rmkraus committed:
Merging balloob/dev into dev.

rmkraus committed:
Added core daemon function as flags.

Created three additional flags for the hass command:
-v -  Toggle verbose log file output
—pid-file -  Specify PID file path
—daemon -  Launch as daemon (nix only)

The core now binds to SIGQUIT on nix systems to trigger a clean
shutdown.

Modified HTTP server to write logging messages through the logging
module.

rmkraus committed:
Merge balloob/dev with rmkraus/daemon

rmkraus commented:
Not quite done yet, just posting for early feedback.

balloob commented on homeassistant/__main__.py:
```diff
@@ -77,8 +77,72 @@ def get_arguments():
+
+    arguments = parser.parse_args()
+    if os.name == "nt":
+        arguments.daemon = False
```
I think that on Windows, arguments.daemon will not exist because of the above if.

rmkraus replied:
Correct, but we need a value for it later. So this will set it for us.

balloob commented on homeassistant/__main__.py:
```diff
@@ -77,8 +77,72 @@ def get_arguments():
+        try:
+            open(pid_file, 'w').write(str(pid))
+        except IOError:
+            pass
```
This seems to me like a fatal error: `Unable to write pid file {}`

rmkraus committed:
Daemon handling cleanup.

More cleanly handling whether a PID file should be checked.
Fatal Error on PID file IOError.
Pylint fix to bootstrap.

rmkraus committed:
Updated Debian daemon script.

rmkraus committed:
Cleanup PID checking logic and write PID logic.

balloob commented on scripts/hass-daemon:
```diff
@@ -0,0 +1,97 @@
+install() {
+    echo "Installing Home Assistant Daemon (hass-daemon)"
+    touch $PID_FILE
+    chwon $RUN_AS $PID_FILE
```
should this be `chown`?

rmkraus committed:
Fixes to new Debian daemon script.
